### DISCUSSION TITLE
Welcome to amazon-dash Discussions!

### DISCUSSION THREAD

**Discussion by Nekmo** [Nekmo/amazon-dash#166] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by chrissv** [comment]
Been using Amazon Dash buttons for general IoT triggers for a couple years. My slightly autistic son has to take medicine in the morning, and his preferred task is getting on the internet. I shut off his internet access (through my dd-wrt wrouter) every evening, and after he takes his meds he pushes the dash button and the internet turns back on (he knows there are dire consequences if he cheats).
I am disappointed that Amazon didn't open source the software (or otherwise allow people to repurpose the buttons). I bought up a dozen buttons on eBay while they are still available (I modified his button so I can replace the battery).
I am a SW nerd with an EE degree. I run openhab2 in my house and love automating things.

**Comment by FHEMUser** [comment]
Hello !
I used it with FHEM and Raspberry Pi3
Today i tested this new version, is a lot more easy like the old.

I have only 3 of this buttons, but i like them.
The MAC is blocked by my router, and if the battery is empty there are a solution to fix it with a sound file.
I made my own buttons with an ESP-07 and deep sleep, but the batteries must be loaded every month :-D

Great projekt ! Many thanks !

**Comment by chrissv** [comment]
It's truely a shame that Amazon hasn't open sourced the code and provided a way for people to repurpose the buttons. Ebay still has [tons of listings ](https://www.ebay.com/sch/i.html?&_nkw=amazon+dash+button&_sacat=See-All-Categories) for Amazon Dash buttons, It would gain Amazon a lot of goodwill (from the geeky folks) if they would do this.
